src/run_quac.py

```diff
@@ -206,6 +210,14 @@ def main(args):
 
     repo_path = Path(get_full_path(__file__)).parents[1]
 
+    # check if dependencies exist
+    for dep_tool in ["singularity", "snakemake"]:
+        if get_tool_path(dep_tool) is None:
+            print(
+                f"ERROR: Tool '{dep_tool}' is required to run QuaC but not found in your environment."
+            )
+            raise SystemExit(1)
+
```

Comment:
```suggestion
    dep_tools = ["singularity", "snakemake"]
    if args.cli_cluster_config or args.snakemake_cluster_config:
        dep_tools.append("sbatch")
        
    for dep_tool in dep_tools:
        if get_tool_path(dep_tool) is None:
            print(
                f"ERROR: Tool '{dep_tool}' is required to run QuaC but not found in your environment."
            )
            raise SystemExit(1)

```
I recommend adding the slurm dependency check here since you're already doing it with other tools and it centralizes the checks to a single place instead of throughout the whole CLI script. This will also make it easier to catch thing like checking if either option requiring slurm's availability has been supplied, in that off chance users run snakemake on the current machine but want jobs to be scheduled via slurm.

Reply:
Good point. I applied this change in a way that print message can be a tiny bit more informative.